components/tours/hooks.ts

```diff
@@ -5,44 +5,72 @@ import {useCallback} from 'react';
 
 import {useDispatch, useSelector} from 'react-redux';
 
+import {without} from 'lodash';
+
 import {getCurrentUserId, isCurrentUserGuestUser} from 'mattermost-redux/selectors/entities/users';
 import {getCurrentRelativeTeamUrl} from 'mattermost-redux/selectors/entities/teams';
+import {getWorkTemplatesLinkedProducts} from 'mattermost-redux/selectors/entities/general';
 
 import {savePreferences} from 'mattermost-redux/actions/preferences';
 import {close as closeLhs, open as openLhs} from 'actions/views/lhs';
 import {setAddChannelDropdown} from 'actions/views/add_channel_dropdown';
 import {switchToChannels} from 'actions/views/onboarding_tasks';
+import {showRHSPlugin} from 'actions/views/rhs';
+import {setProductMenuSwitcherOpen} from 'actions/views/product_menu';
+
+import {useGetRHSPluggablesIds} from 'components/work_templates/hooks';
+
 import {getHistory} from 'utils/browser_history';
+import {suitePluginIds} from 'utils/constants';
+import {useProducts} from 'utils/products';
 import {GlobalState} from 'types/store';
 
 import {OnboardingTaskCategory, OnboardingTaskList, OnboardingTasksName} from '../onboarding_tasks';
 
-import {setProductMenuSwitcherOpen} from '../../actions/views/product_menu';
-
 import {
     CrtTutorialSteps,
     ExploreOtherToolsTourSteps,
     FINISHED,
     OnboardingTourSteps,
     TTNameMapToTourSteps,
     TutorialTourName,
+    WorkTemplateTourSteps,
 } from './constant';
 
 export const useGetTourSteps = (tourCategory: string) => {
     const isGuestUser = useSelector((state: GlobalState) => isCurrentUserGuestUser(state));
     const pluginsList = useSelector((state: GlobalState) => state.plugins.plugins);
-
+    const pluginProducts = useProducts();
+    const workTemplatesLinkedItems = useSelector((state: GlobalState) => getWorkTemplatesLinkedProducts(state));
     let tourSteps: Record<string, number> = TTNameMapToTourSteps[tourCategory];
 
+    let boardsProductEnabled = false;
+    let playbooksProductEnabled = false;
+    if (pluginProducts) {
+        boardsProductEnabled = pluginProducts.some((product) => (product.pluginId === suitePluginIds.focalboard) || (product.pluginId === suitePluginIds.boards));
+        playbooksProductEnabled = pluginProducts.some((product) => product.pluginId === suitePluginIds.playbooks);
+    }
```

Comment:
Wondering if we should move this check to a shareable selector as we may start repeating this in several places

Reply:
done